src/main/cluster.ts

```diff
@@ -643,17 +647,26 @@ export class Cluster implements ClusterModel, ClusterState {
       if (!this.allowedNamespaces.length) {
         return [];
       }
-      const resourceAccessStatuses = await Promise.all(
-        apiResources.map(apiResource => this.canI({
-          resource: apiResource.resource,
-          group: apiResource.group,
-          verb: "list",
-          namespace: this.allowedNamespaces[0]
-        }))
-      );
+
+      const resources = apiResources.filter((resource) => this.resourceAccessStatuses.get(resource) === undefined);
+
+      for (const apiResource of resources) {
+        for (const namespace of this.allowedNamespaces.slice(0, 10)) {
+          if (!this.resourceAccessStatuses.get(apiResource)) {
+            const result = await this.canI({
```

Comment:
Why not parallel loading? Because of it's caching it doesn't matter?

Reply:
Just to not cause too many parallel request to k8s api, these requests might be heavy (depending on Kubernetes distro / config).

Reply:
Can we specialize this? Namely, for the time being have MKE do it serially and others do it up to 10 requests at a time?

Reply:
@Nokel81 no we can't really. I tested this on MKE but we don't really know how each Kubernetes cluster is configured.